common/functions/src/scalars/strings/locate.rs

```diff
@@ -0,0 +1,231 @@
+        Some((2, 3))
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(true)
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::UInt64)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let ss_column = columns[0].column().cast_with_type(&DataType::String)?;
+        let s_column = columns[1].column().cast_with_type(&DataType::String)?;
+        let p_column = if columns.len() == 3 {
+            columns[2].column().cast_with_type(&DataType::UInt64)?
+        } else {
+            DataColumn::Constant(DataValue::UInt64(Some(1)), input_rows)
+        };
+
+        let r_column: DataColumn = match (ss_column, s_column, p_column) {
+            (
+                DataColumn::Constant(DataValue::String(ss), _),
+                DataColumn::Constant(DataValue::String(s), _),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(ss), Some(s), Some(p)) = (ss, s, p) {
+                    DataColumn::Constant(DataValue::UInt64(Some(find_at(&s, &ss, &p))), input_rows)
+                } else {
+                    DataColumn::Constant(DataValue::Null, input_rows)
+                }
+            }
+            (
+                DataColumn::Constant(DataValue::String(ss), _),
+                DataColumn::Array(s_series),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(ss), Some(p)) = (ss, p) {
+                    DFUInt64Array::from_iter(
+                        s_series
+                            .string()?
+                            .into_no_null_iter()
+                            .map(|s| find_at(s, &ss, &p)),
+                    )
+                    .into()
+                } else {
+                    DataColumn::Constant(DataValue::Null, input_rows)
+                }
+            }
+            (
+                DataColumn::Array(ss_series),
+                DataColumn::Constant(DataValue::String(s), _),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(s), Some(p)) = (s, p) {
+                    DFUInt64Array::from_iter(
+                        ss_series
+                            .string()?
+                            .into_no_null_iter()
+                            .map(|ss| find_at(&s, ss, &p)),
+                    )
+                    .into()
+                } else {
+                    DataColumn::Constant(DataValue::Null, input_rows)
+                }
+            }
+            (
+                DataColumn::Array(ss_series),
+                DataColumn::Array(s_series),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let Some(p) = p {
+                    DFUInt64Array::from_iter(
+                        ss_series
```

Comment:
`into_no_null_iter` ignore the null values.

We can use `DFUInt64Array::new_from_iter_validity` to extends the validity of `ss_series`.

Reply:
Fixed...